File: FragmentExample/app/src/main/java/kobe/fragmentexample/MainActivity.java
package kobe.fragmentexample;

import android.net.Uri;
import android.os.Bundle;
import androidx.fragment.app.FragmentManager;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements ContentFragment.OnFragmentInteractionListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //>> load Fragment dynamically
        //>> if you don't want load Fragment dynamically
        //>> you can use fragment tag in the layout XML
        FragmentManager fragmentManager = getSupportFragmentManager();
        fragmentManager.beginTransaction()
                .add(R.id.fragment_place, ContentFragment.newInstance("a","b"))
                .commit();
    }

    @Override
    public void onFragmentInteraction(Uri uri) {

    }
}

File: Kotlin_FragmentExample/app/src/main/java/com/intowow/kotlin_fragmentexample/MainActivity.kt
package com.intowow.kotlin_fragmentexample

import android.net.Uri
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentTransaction
import android.support.v7.app.AppCompatActivity


class MainActivity : AppCompatActivity(), ContentFragment.OnFragmentInteractionListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //>> now, you only need to use addFragment() to add Fragment dynamically
        addFragment(R.id.fragment_place, ContentFragment.newInstance("a","b"))
    }

    override fun onFragmentInteraction(uri: Uri) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    //>> extension function
    inline fun FragmentManager.inTransation(func: FragmentTransaction.() -> FragmentTransaction) {
        beginTransaction().func().commit()
    }

    //>> extension function
    fun AppCompatActivity.addFragment(id: Int, fragment: Fragment) {
        supportFragmentManager.inTransation { add(id, fragment) }
    }
}
